src/cascadia/TerminalSettingsModel/ActionArgs.cpp

```diff
@@ -709,23 +709,40 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 
     winrt::hstring SuggestionsArgs::GenerateName() const
     {
-        auto base{ RS_(L"SuggestionsCommandKey") };
-        switch (Source())
+        std::wstringstream ss;
+        ss << RS_(L"SuggestionsCommandKey").c_str();
+
+        if (UseCommandline())
         {
-        case SuggestionsSource::CommandHistory:
-            base = RS_(L"SuggestionsCommandHistoryCommandKey");
+            ss << L", useCommandline:true";
         }
 
-        if (UseCommandline())
+        // All of the source values will leave a trailing ", " that we need to chop later:
+        ss << L", source: ";
+        const auto source = Source();
+        if (source == SuggestionsSource::All)
         {
-            return winrt::hstring{
-                fmt::format(L"{}, useCommandline:true", std::wstring_view(base))
-            };
+            ss << L"all, ";
+        }
+        else if (source == static_cast<SuggestionsSource>(0))
+        {
+            ss << L"none, ";
         }
         else
         {
-            return base;
+            if (WI_IsFlagSet(source, SuggestionsSource::Tasks))
+            {
+                ss << L"tasks, ";
+            }
+
+            if (WI_IsFlagSet(source, SuggestionsSource::CommandHistory))
+            {
+                ss << L"commandHistory, ";
+            }
         }
+        // Chop off the last ","
+        auto result = ss.str();
+        return winrt::hstring{ result.substr(0, result.size() - 2) };

```

Comment:
nit: `result.resize(result.size() - 2)` will avoid duplicating the entire string. (`substr` doesn't create a view.)